### DISCUSSION TITLE
Is it possible to use a c++ middleware in actix-web?

### DISCUSSION THREAD

**Discussion by dns2utf8-tipi** [actix/actix-web#3128] [Ideas]
Hi all

The problem I face is that there is some complicated authentication code in a legacy application that I would like to reuse in a new actix-web based application.
Is there a plugin that can hook the FFI or would I have to build my own wrapper?

Thanks in advance :)

**Comment by asonix** [comment]
The answer is very technically "yes this is possible" but you would need to write a lot of the FFI bits yourself most likely. You may want to check out `bindgen` https://rust-lang.github.io/rust-bindgen/cpp.html
